File: Bitwise/78_subsets/main.py
from typing import List


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        mask = 0
        result_nums: List[List[int]] = []
        while mask < (1 << len(nums)):
            sub_nums = []
            for i in range(len(nums)):
                if (mask & (1 << i)):
                    sub_nums.append(nums[i])
            result_nums.append(sub_nums)
            mask += 1
        return result_nums

File: Bitwise/78_subsets/main.go
package main

func subsets(nums []int) [][]int {
	resultNums := make([][]int, 0, 1<<len(nums))
	for mask := 0; mask < (1 << len(nums)); mask++ {
		innerNums := []int{}
		for idx, num := range nums {
			if (mask & (1 << idx)) > 0 {
				innerNums = append(innerNums, num)
			}
		}
		resultNums = append(resultNums, innerNums)
	}
	return resultNums
}
